packages/opentelemetry-core/src/context/propagation/HttpTraceContext.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { SpanContext, HttpTextFormat } from '@opentelemetry/types';
+import { TraceState } from '../../trace/TraceState';
+
+export const TRACE_PARENT_HEADER = 'traceparent';
+export const TRACE_STATE_HEADER = 'tracestate';
+// TODO: https://github.com/open-telemetry/opentelemetry-js/pull/60
+// Consider to import const from TraceOptions.UNSAMPLED
+export const DEFAULT_OPTIONS = 0x0;
+const VALID_TRACE_PARENT_REGEX = /^[\da-f]{2}-[\da-f]{32}-[\da-f]{16}-[\da-f]{2}$/;
+const VALID_TRACEID_REGEX = /^[0-9a-f]{32}$/i;
+const VALID_SPANID_REGEX = /^[0-9a-f]{16}$/i;
+const VALID_ID_REGEX = /^0+$/i;
+const VERSION = '00';
+
+// TODO: Consider to move these interfaces in otel-types.
+export interface HeaderGetter {
```

Comment:
Carrier should be any object. Why require a specific class?

Reply:
I think I've addressed everything. PTAL